frontend/components/Layout/Navigation.tsx

```diff
@@ -0,0 +1,406 @@
+import {
+   Accordion,
+   AccordionButton,
+   AccordionIcon,
+   AccordionItem,
+   AccordionPanel,
+   Box,
+   BoxProps,
+   Divider,
+   Drawer,
+   DrawerBody,
+   DrawerContent,
+   DrawerOverlay,
+   Flex,
+   FlexProps,
+   forwardRef,
+   Icon,
+   IconButton,
+   IconButtonProps,
+   useDisclosure,
+} from '@chakra-ui/react';
+import { LayoutData } from '@lib/api';
+import NextLink from 'next/link';
+import * as React from 'react';
+import { HiArrowNarrowRight } from 'react-icons/hi';
+
+type NavigationMenu = NonNullable<LayoutData['header']['menu']>;
+
+export interface NavigationProps {
+   menu: NavigationMenu;
+}
+
+const ANIMATION_DURATION = '300ms';
+
+const DesktopNavigation = forwardRef<BoxProps & NavigationProps, 'nav'>(
+   ({ menu, ...boxProps }, ref) => {
+      return (
+         <Box ref={ref} as="nav" h="full" ml="6" {...boxProps}>
+            <MenuBar h="full" position="relative">
+               {menu.items.map((item, index) => {
+                  switch (item.type) {
+                     case 'submenu': {
+                        return (
+                           <MenuItem
+                              key={index}
+                              transition="background-color 300ms"
```

Comment:
Not sure if this was intentional, but this transition could use `ANIMATION_DURATION` from above